src/my-bot.ts

```diff
@@ -1,38 +1,65 @@
+import {Position} from './Position'
+import {Matrix} from './matrixClass'
+import {TargetingMethods} from './targeting'
+import {MyShips} from './myShips'
+
 export class MyBot {
+    private state;
     public getShipPositions() {
-        return [
-            { StartingSquare: { Row: "A", Column: 1 }, EndingSquare : { Row: "A", Column: 5 } },
-            { StartingSquare: { Row: "C", Column: 1 }, EndingSquare : { Row: "C", Column: 4 } },
-            { StartingSquare: { Row: "E", Column: 1 }, EndingSquare : { Row: "E", Column: 3 } },
-            { StartingSquare: { Row: "G", Column: 1 }, EndingSquare : { Row: "G", Column: 3 } },
-            { StartingSquare: { Row: "I", Column: 1 }, EndingSquare : { Row: "I", Column: 2 } },
-        ]
-    }
+        let myShips = new MyShips([]);
+        let result = myShips.createBoard();
+        console.log(result);
+        return result;
 
-    public selectTarget(gamestate) {
-        var previousShot = gamestate.MyShots && gamestate.MyShots[gamestate.MyShots.length-1];
-        if(previousShot) {
-            return this.getNextTarget(previousShot.Position);
-        }
-        return { Row: "A", Column: 1 };  
+     
     }
 
-    private getNextTarget(position) {
-        var column = this.getNextColumn(position.Column);
-        var row = column === 1 ? this.getNextRow(position.Row) : position.Row;
-        return { Row: row, Column: column }
-    }
+    public selectTarget(gameState) {
+        try{
+            console.log("turn "+ (gameState.MyShots.length+1))
+            var mat:Matrix = new Matrix(gameState);
+            var ispreviousShot = gameState.MyShots && gameState.MyShots[gameState.MyShots.length-1];
+            if(ispreviousShot) {
+                var previousShot = gameState.MyShots[gameState.MyShots.length-1];
+                var previousShotAsCorrectClass = new Position(previousShot.Position.Row, previousShot.Position.Column);
+                if(previousShot.WasHit)
+                {
+                    if(!mat.isThereUnsunkShipAt(previousShotAsCorrectClass)) return this.getNextTarget(mat, gameState);
+                    else {
+                        // console.log(mat.board);
+                        return mat.returnUnsunkShot();
+                    }
+                }
+                else{
+                    for(let i = 1; i <= gameState.MyShots.length; i ++){
+                        if(gameState.MyShots[gameState.MyShots.length-i].WasHit){
+                            //this is the most recent hit;
+                            if(!mat.isThereUnsunkShipAt(new Position(gameState.MyShots[gameState.MyShots.length-i].Position.Row, gameState.MyShots[gameState.MyShots.length-i].Position.Column))) return this.getNextTarget(mat, gameState);
+                            else{
+                                // console.log(mat.board);

```

Comment:
Don't particularly like this being commented out, but I know why.

In an actual project, we'd probably set this logging to `debug` only, and then have some config setting to turn it on/off,. but that's too much for a 3 day battleship bot.